### PR TITLE
Start Sauce Connect Tunnel if provided tunnelIdentifier is not found

### PR DESCRIPTION
**Pull request by christian-bromann**
It would be really helpful if Speedo could start the tunnel for the user in case the tunnel hasn't been started manually. For that we need to check the API if such tunnel identifier already exists and then use that or start a tunnel manually.

### CODE REVIEW

File: src/commands/run.js [hunk 1]
```diff
@@ -194,6 +215,17 @@ export const handler = async (argv) => {
         status.stopAndPersist({ text: 'continuing ...' })
     }
 
+    /**
+     * stop tunnel if one was created in the beginning
+     */
+    if (tunnelProcess) {
+        status.start('Stopping Sauce Connect tunnel...')
+        await new Promise((resolve) => tunnelProcess.close(() => {
```
Comment (adambabik) [inline src/commands/run.js]: Any timeout needed? What if this errors? Possibly leaking tunnel processes?

Comment (adambabik) [inline src/commands/run.js]: Any timeout needed? What if this errors? Possibly leaking tunnel processes?

File: tests/run.test.js [hunk 2]
```diff
@@ -128,6 +128,50 @@ test('should run successfully', async () => {
     expect(ora().stopAndPersist.mock.calls).toMatchSnapshot()
 })
 
+test.only('should run successfully with tunnels', async () => {
```
Comment (adambabik) [inline tests/run.test.js]: `only`

File: tests/run.test.js [hunk 3]
```diff
@@ -128,6 +128,50 @@ test('should run successfully', async () => {
...
+        'bar',
+        '/some/tmpDir',
+        opts,
+        expect.any(Object)
+    )
+    expect(tunnelMock.close).toBeCalledTimes(1)
+    startTunnel.mockClear()
+})
+
+test.only('should not close tunnel if none was started', async () => {
```
Comment (adambabik) [inline tests/run.test.js]: `only`

File: src/commands/run.js [hunk 4]
```diff
@@ -194,6 +218,21 @@ export const handler = async (argv) => {
         status.stopAndPersist({ text: 'continuing ...' })
     }
 
+    /**
+     * stop tunnel if one was created in the beginning
+     */
+    if (tunnelProcess) {
+        status.start('Stopping Sauce Connect tunnel...')
+        await new Promise((resolve) => {
+            const shutdownTimeout = setTimeout(resolve, TUNNEL_SHUTDOWN_TIMEOUT)
```
Comment (adambabik) [inline src/commands/run.js]: Shouldn't it reject in such a case?

Comment (christian-bromann) [reply src/commands/run.js]: It is highly unlikely that this fails or times-out. If so we can just ignore the fact and should not let the test fail.

Comment (adambabik) [reply src/commands/run.js]: Yeah, it should not fail the test but maybe logging that would be useful. Then one can clean up the tunnel process by themselves.

Comment (christian-bromann) [reply src/commands/run.js]: good call

**Pull request merged**